File: js/2574.0_Left_and_Right_Sum_Differences.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var leftRightDifference = function(nums) {
    const total = _.sum(nums);
    let leftSum = 0;
    for (let i = 0; i < nums.length; i++) {
        const x = nums[i];
        nums[i] = Math.abs(leftSum * 2 + x - total);
        leftSum += x;
    }
    return nums;
};

File: java/2574.0_Left_and_Right_Sum_Differences.java
class Solution {
    public int[] leftRightDifference(int[] nums) {
        int total = 0;
        for (int x: nums) {
            total += x;
        }

        int leftSum = 0;
        for (int i = 0; i < nums.length; i++) {
            int x = nums[i];
            nums[i] = Math.abs(leftSum * 2 + x - total);
            leftSum += x;
        }
        return nums;
    }
}
